scipy/sparse/linalg/_dsolve/_superluobject.c

```diff
@@ -105,10 +105,76 @@ static PyObject *SuperLU_solve(SuperLUObject * self, PyObject * args,
     return NULL;
 }
 
+
+static PyObject *SuperLU_rinvnormest(SuperLUObject * self, PyObject * args,
+                               PyObject * kwds)
+{
+    volatile int info;
+    volatile double rcond;
+    const char* volatile norm = "1";
+    char norm_c;
+    volatile SuperLUStat_t stat = { 0 };
+    static char *kwlist[] = { "norm", NULL };
+    volatile jmp_buf *jmpbuf_ptr;
+    SLU_BEGIN_THREADS_DEF;
+
+    if (!CHECK_SLU_TYPE(self->type)) {
+        PyErr_SetString(PyExc_ValueError, "unsupported data type");
+        return NULL;
```

Comment:
We need a test that this error is indeed raised. Same for the other ValueError below.

Reply:
Some of the errors might be unreachable because  the SuperLU decomposition would fail before the rinvnorm function is called, but I've added a check for the `norm` argument in [d43b143](https://github.com/scipy/scipy/pull/21620/commits/d43b1438f53a4c31046a42955f9a56150d814af6)